source/application/AMPIC887DriverServer/AMPIC887HexapodState.cpp

```diff
@@ -0,0 +1,454 @@
+			boolString = "Yes";
+		} else {
+			boolString = "No";
+		}
+		stateString.append(QString("Soft Limits Active: %1\n")
+						   .arg(boolString));
+		if(limitSwitchStates_.value(currentAxis)) {
+			boolString = "Yes";
+		} else {
+			boolString = "No";
+		}
+		stateString.append(QString("Hard Limits Active: %1\n")
+						   .arg(boolString));
+		stateString.append(QString("Step Size: %1 %2\n")
+						   .arg(stepSizes_.value(currentAxis))
+						   .arg(units));
+		stateString.append(QString("Min. Position: %1 %2\n")
+						   .arg(minCommandablePositions_.value(currentAxis))
+						   .arg(units));
+		stateString.append(QString("Max. Position: %1 %2\n")
+						   .arg(maxCommandablePositions_.value(currentAxis))
+						   .arg(units));
+
+		if(currentAxis == AMGCS2::XAxis ||
+				currentAxis == AMGCS2::YAxis ||
+				currentAxis == AMGCS2::ZAxis) {
+			stateString.append(QString("Pivot Point: %1 %2\n")
+							   .arg(pivotPoints_.value(currentAxis))
+							   .arg(units));
+		}
+
+	}
+
+	return stateString.trimmed();
+}
+
+bool AMPIC887HexapodState::areAllAxesReferenced() const
+{
+	return isInitialized_ &&
+			referencedStates_.value(AMGCS2::XAxis) &&
+			referencedStates_.value(AMGCS2::YAxis) &&
+			referencedStates_.value(AMGCS2::ZAxis) &&
+			referencedStates_.value(AMGCS2::UAxis) &&
+			referencedStates_.value(AMGCS2::VAxis) &&
+			referencedStates_.value(AMGCS2::WAxis);
+}
+
+bool AMPIC887HexapodState::referencedState(AMGCS2::Axis axis) const
+{
+	return referencedStates_.value(axis, false);
+}
+
+const AMPIC887AxisMap<bool> AMPIC887HexapodState::referencedStates() const
+{
+	return referencedStates_;
+}
+
+void AMPIC887HexapodState::setReferencedState(AMGCS2::Axis axis, bool referencedState)
+{
+	if(axis == AMGCS2::UnknownAxis) {
+		return;
+	}
+
+	referencedStates_.insert(axis, referencedState);
+}
+
+void AMPIC887HexapodState::setReferencedStates(const AMPIC887AxisMap<bool> &referencedStates)
+{
+	AMPIC887AxisCollection axes = referencedStates.axes();
+	foreach(AMGCS2::Axis currentAxis, axes) {
+
+		referencedStates_.insert(currentAxis, referencedStates.value(currentAxis));
+	}
+}
+
+double AMPIC887HexapodState::currentPosition(AMGCS2::Axis axis) const
+{
+	return currentPositions_.value(axis, 0);
+}
+
+const AMPIC887AxisMap<double> AMPIC887HexapodState::currentPositions() const
+{
+	return currentPositions_;
+}
+
+void AMPIC887HexapodState::setCurrentPosition(AMGCS2::Axis axis, double currentPosition)
+{
+	if(axis == AMGCS2::UnknownAxis) {
+		return;
+	}
+
+	currentPositions_.insert(axis, currentPosition);
+}
+
+void AMPIC887HexapodState::setCurrentPositions(const AMPIC887AxisMap<double> &currentPositions)
+{
+	AMPIC887AxisCollection axes = currentPositions.axes();
+	foreach(AMGCS2::Axis currentAxis, axes) {
+
+		currentPositions_.insert(currentAxis, currentPositions.value(currentAxis));
+	}
+}
+
+double AMPIC887HexapodState::targetPosition(AMGCS2::Axis axis) const
+{
+	return targetPositions_.value(axis, 0);
```

Comment:
Will it cause any errors if you somehow returned 0 when you didn't want to because the axis wasn't valid?